easybuild/easyblocks/c/cudacompat.py

```diff
@@ -0,0 +1,133 @@
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for CUDA compat libraries, implemented as an easyblock
+
+Ref: https://docs.nvidia.com/deploy/cuda-compatibility/index.html#manually-installing-from-runfile
+
+@author: Alexander Grund (TU Dresden)
+"""
+
+import os
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.generic.binary import Binary
+from easybuild.framework.easyconfig import MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import copy_file, find_glob_pattern, mkdir, symlink
+from easybuild.tools.run import run_cmd
+
+
+class EB_CUDAcompat(Binary):
+    """
+    Support for installing CUDA compat libraries.
+    """
+
+    @staticmethod
+    def extra_options():
+        """Add variable for driver version"""
+        extra_vars = Binary.extra_options()
+        extra_vars.update({
+            'nv_version': [None, "Version of the driver package", MANDATORY],
+        })
+        # We don't need the extract and install step from the Binary EasyBlock
+        del extra_vars['extract_sources']
+        del extra_vars['install_cmd']
+        # And also no need to modify the PATH
+        del extra_vars['prepend_to_path']
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Initialize custom class variables for Clang."""
+        super(EB_CUDAcompat, self).__init__(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract the files without running the installer."""
+        execpath = self.src[0]['path']
+        tmpdir = os.path.join(self.builddir, 'tmp')
+        targetdir = os.path.join(self.builddir, 'extracted')
+        run_cmd("/bin/sh " + execpath + " --extract-only --tmpdir='%s' --target '%s'" % (tmpdir, targetdir))
+        self.src[0]['finalpath'] = targetdir
+
+    def install_step(self):
+        """Install CUDA compat libraries by copying library files and creating the symlinks."""
+        libdir = os.path.join(self.installdir, 'lib')
```

Comment:
```suggestion
        # EULA for NVIDIA driver must be accepted via --accept-eula-for EasyBuild configuration option,
        # or via 'accept_eula = True' in easyconfig file
        self.check_accepted_eula(
            more_info='https://www.nvidia.com/content/DriverDownload-March2009/licence.php?lang=us'
        )
        
        libdir = os.path.join(self.installdir, 'lib')
```

Reply:
Done but added "NVIDIA-driver" as the name similar to "Intel-oneAPI"

Also added this to the `fetch_step` as the (skipped by EB) download popup reads
> By clicking the "Agree & Download" button below, you are confirming that you have read and agree to be bound by the [License For Customer Use of NVIDIA Software](https://www.nvidia.com/content/DriverDownload-March2009/licence.php?lang=us) for use of the driver.